pkg/ebpf/c/common/filtering.h

```diff
@@ -168,6 +170,25 @@ statfunc u64 bool_filter_matches(u64 match_bitmap, bool bool_value)
     return match_bitmap ^ (bool_value ? ~0ULL : 0);
 }
 
+statfunc void reverse_string(char *dst, char *src, int len)
+{
+    uint i;
+
+    // don't count null-termination since we will force it at the end
+    len = (len - 1) & MAX_DATA_FILTER_STR_SIZE_MASK;
+
+    // Copy with safe bounds checking
+    for (i = 0; i < len; i++) {
+        if (i >= MAX_DATA_FILTER_STR_SIZE) {
+            break;
+        }
```

Comment:
We should break or return? Currently this writes nul char to dst even if len is 0.

nit: only use curly braces in blocks with more than one line or if the else block is not onelined.

```suggestion
        if (i >= MAX_DATA_FILTER_STR_SIZE)
            break;
```